src/pcp/types.rs

```diff
@@ -7,6 +7,12 @@ use crate::pcp::gadgets::{Mul, PolyEval};
 use crate::pcp::{GadgetCallOnly, PcpError, Value};
 use crate::polynomial::poly_range_check;
 
+use std::convert::TryFrom;
+
+/// Errors propagagted by methods in this module.
+#[derive(Debug, PartialEq, thiserror::Error)]
+pub enum TypeError {}
```

Comment:
It looks like there are currently no failure modes for any of the `Value` implementations. Maybe we could use [`std::convert::Infallible`](https://doc.rust-lang.org/std/convert/enum.Infallible.html) here? Each `Value` implementation's `TryFrom` implementation can use a different associated error type, so if/when a `Value` comes along whose `TryFrom` can fail, it can provide an error enum.

Reply:
Done. (There is a failure case for a data type that will be coming in a future PR.)